packages/SwingSet/src/kernel/vatManager/manager-subprocess-xsnap.js

```diff
@@ -96,8 +96,8 @@ export function makeXsSubprocessFactory({
       }
     }
 
-    /** @type { (msg: Uint8Array) => Uint8Array } */
-    function handleCommand(msg) {
+    /** @type { (msg: Uint8Array) => Promise<Uint8Array> } */
+    async function handleCommand(msg) {
```

Comment:
@warner thanks for helping me narrow the search for the bug with the xs worker test and `panic()`. This `async` seems to be the fix.

Oops... this commit is more than docs then. Will fix in due course (perhaps by squashing this PR).

Reply:
Huh, I don't understand why adding `async` there helps.. I can imagine that it changes what the caller sees. Previously, `handleCommand()` might throw, or return a non-promise. With `async`, it always returns a promise, which either rejects or resolves. The one place I can find that calls `handleCommand` is in `xsnap.js`:

https://github.com/Agoric/agoric-sdk/blob/069201d693e8f9ed42eaa99cc998c937ce9e82a5/packages/xsnap/src/xsnap.js#L196-L198

and from what I can tell, the `await` means that returning a rejected promise will reject the enclosing function, and the `async` on the enclosing function) means that any exception thrown by (a non-`async`) `handleCommand` will do the same thing. So I'm not sure what putting `async` on `handleCommand` could change.

OTOH, I remember part of the problem was Node's "unhandled rejected promise" warning, and maybe it's enough to change the point at which a rejection-handler gets added, which might hush the warning.

Reply:
yes, this addressed the unhandled rejection, I think.